test/MSBuildBinLogQuery.CommandLine/Program.cs

```diff
@@ -21,37 +22,79 @@ public static void Main(string[] args)
                 PrintErrorMessage($"File not found: {args[0]}");
             }
 
-            var events = new BinaryLogReader(args[0]).ReadEvents();
+            using var binaryLogReader = new BinaryLogReader(args[0]);
+            var events = binaryLogReader.ReadEvents();
 
             var graphBuilder = new GraphBuilder();
             graphBuilder.HandleEvents(events.ToArray());
 
-            PrintProjectNodes(graphBuilder.Projects.Values);
+            PrintProjectNodes(graphBuilder.Projects.Values.ToList());
```

Comment:
Since it `PrintProjectNodes` takes `IEnumerable`, there's no need to make it a list first; doing so can result in unnecessary allocations. Doesn't matter here, more important as a general principle.

Reply:
Done.